probe-rs/src/debug/variable_cache.rs

```diff
@@ -9,6 +10,58 @@ pub struct VariableCache {
     pub(crate) variable_hash_map: HashMap<i64, Variable>,
 }
 
+impl Serialize for VariableCache {
```

Comment:
Can we absolutely not do this with the standard derive macros?

Reply:
I don't believe so. This takes a hashmap of items and constructs a hierarchical tree view of variables and their children.